### ISSUE TITLE
[Bug] -b flag fails when no results match query

### ISSUE THREAD

**Issue by markmsmith** [harelba/q#131]
Hey, thanks for this awesome tool, it's made my life much easier.  Just thought I'd let you know about the following issue when you try to use the `-b` flag.

With the input files  
test1.csv:

```
heading1,heading2
value1,value2
```

test2.csv:

```
heading1,heading2
value3,value4
```

If you run the following query:  
`q -HOb -d, "select a.heading1, b.heading2 from ./test1.csv a join ./test2.csv b on a.heading1 = b.heading1"`  
You should get no results, since the values under heading1 don't match, but instead you get the error:

```
Traceback (most recent call last):
  File "/usr/local/bin/q", line 1639, in <module>
    run_standalone()
  File "/usr/local/bin/q", line 1628, in run_standalone
    q_output_printer.print_output(STDOUT,sys.stderr,q_output)
  File "/usr/local/bin/q", line 1319, in print_output
    self._print_output(f_out,f_err,results)
  File "/usr/local/bin/q", line 1361, in _print_output
    fmt_str = "%%-%ss" % max_lengths[i]
IndexError: list index out of range
```

Removing the `-b` flag correctly gives just the expected output of just the headings:

```
heading1,heading2
```

Similarly, if you change test2.csv to match:

```
heading1,heading2
value1,value4
```

then the -b flag will work again:

```
q -HOb -d, "select a.heading1, b.heading2 from ./test1.csv a join ./test2.csv b on a.heading1 = b.heading1"
heading1,heading2
value1,value4
```

so it seems to only happen if there's no results.

**Comment by serima** [comment]
I installed q via homebrew.

```
% brew info q
q: stable 1.5.0, HEAD
Treat text as a database
https://github.com/harelba/q
/usr/local/Cellar/q/1.5.0 (4 files, 69.6K) *
  Built from source on 2016-11-16 at 11:33:52
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/q.rb
```

Reproduced it.

```
% cat test1.csv
heading1,heading2
value1,value2
% cat test2.csv
heading1,heading2
value3,value4
% q -HOb -d, "select a.heading1, b.heading2 from ./test1.csv a join ./test2.csv b on a.heading1 = b.heading1"
Traceback (most recent call last):
  File "/usr/local/bin/q", line 1639, in <module>
    run_standalone()
  File "/usr/local/bin/q", line 1628, in run_standalone
    q_output_printer.print_output(STDOUT,sys.stderr,q_output)
  File "/usr/local/bin/q", line 1319, in print_output
    self._print_output(f_out,f_err,results)
  File "/usr/local/bin/q", line 1361, in _print_output
    fmt_str = "%%-%ss" % max_lengths[i]
IndexError: list index out of range
zsh: exit 1     q -HOb -d,
% q --version
q version 1.5.0
Copyright (C) 2012-2014 Harel Ben-Attia (harelba@gmail.com, @harelba on twitter)
http://harelba.github.io/q/
```

Then I cloned this repo and tried same thing.

```
% git show --summary
commit cfe2d047810cc54821bc3832e2076d373c643d0a
Author: Harel Ben-Attia <harelba@gmail.com>
Date:   Mon Jan 30 01:06:10 2017 +0200

    Update README.markdown

% git br
* master
% ./bin/q -HOb -d, "select a.heading1, b.heading2 from ./test1.csv a join ./test2.csv b on a.heading1 = b.heading1"
heading1,heading2
% ./bin/q --version
q version 1.6.0notreleasedyet
Copyright (C) 2012-2014 Harel Ben-Attia (harelba@gmail.com, @harelba on twitter)
http://harelba.github.io/q/
```

This bug seems to be fixed in Ver 1.6.0. this commit? https://github.com/harelba/q/commit/a053b815312c0d8eadf6f42b47ddd6edbe1e4fb8

**Comment by svenXY** [comment]
Great that it's fixed, I can confirm it.

So - when do you plan to release 1.6.0 for Arch? There is another bug that you fixed as well for me since then...

Thanks,
Sven

**Comment by serima** [comment]
@harelba I also want to know when is the next release for brew packages.

**Comment by harelba** [comment]
Fixed in v1.6.3. 

@svenXY I'm sorry, the arch linux version is still 1.5.0 - Someone else created the arch linux version for 1.5.0, and I'm not familiar with the process. If you still require it, ping me, and I'll try to dig deeper.

**Comment by svenXY** [comment]
I put a comment into the AUR page for q.
Thanks for fixing it anyway.

Sven
